js_modules/dagster-ui/packages/ui-core/src/runs/CapturedLogPanel.tsx

```diff
@@ -33,28 +35,85 @@ export const CapturedOrExternalLogPanel = React.memo(
+      (props.visibleIOType === 'stdout'
+        ? logCaptureInfo.stdoutShellCmd
+        : logCaptureInfo.stderrShellCmd);
+
+    const copy = useCopyToClipboard();
+    const onClickFn = async (key: string, value: string | undefined) => {
+      if (!value) {
+        return;
+      }
+      copy(value);
+      await showSharedToaster({
+        intent: 'success',
+        icon: 'done',
+        message: `${key} string copied!`,
+      });
+    };
+    const onClickExternalUri = async () => onClickFn('log artefact URI', uriOrPath);
+    const onClickShellCmd = async () => onClickFn('shell command', shellCmd);
+
+    if (externalUrl || uriOrPath || shellCmd) {
       return (
         <Box
-          flex={{direction: 'row', alignItems: 'center', justifyContent: 'center', gap: 1}}
+          flex={{direction: 'column', alignItems: 'center', justifyContent: 'center', gap: 1}}
           background={Colors.tooltipBackground()}
           style={{color: Colors.tooltipText(), flex: 1, minHeight: 0}}
         >
-          View logs at
-          <a
-            href={externalUrl}
-            target="_blank"
-            rel="noreferrer"
-            style={{
-              color: Colors.tooltipText(),
-              textDecoration: 'underline',
-              marginLeft: 4,
-              marginRight: 4,
-            }}
-          >
-            {externalUrl}
-          </a>
-          <Icon name="open_in_new" color={Colors.tooltipText()} size={20} style={{marginTop: 2}} />
+          {externalUrl ? (
+            <div>
+              View logs at
+              <a
+                href={externalUrl}
+                target="_blank"
+                rel="noreferrer"
+                style={{
+                  color: Colors.tooltipText(),
+                  textDecoration: 'underline',
+                  marginLeft: 4,
+                  marginRight: 4,
+                }}
+              >
+                {externalUrl}
+              </a>
+              <Icon
+                name="open_in_new"
+                color={Colors.tooltipText()}
+                size={20}
+                style={{marginTop: 2}}
+              />
+            </div>
+          ) : undefined}
+
+          {uriOrPath ? (
+            <div>
+              Logs artefact URI:
+              <Tooltip content="Click to copy log artefact URI" placement="top" display="block">
+                <UnstyledButton onClick={onClickExternalUri}>{uriOrPath}</UnstyledButton>
+              </Tooltip>
+            </div>
+          ) : undefined}
+
+          {shellCmd ? (
+            <div>
+              Shell command to download logs:
+              <Tooltip
+                content="Click to copy shell command to download logs"
+                placement="top"
+                display="block"
+              >
+                <UnstyledButton onClick={onClickShellCmd}>{shellCmd}</UnstyledButton>
+              </Tooltip>
+            </div>
+          ) : undefined}
```

Comment:
With a table, rows can look like this:

```
{shellCmd ? (
  <tr>
    <td>Shell command to download logs</td>
    <td>
      <Tooltip content="..." placement="top">
        <UnstyledButton onClick={...}>
          <Mono>{shellCmd}</Mono>
        </UnstyledButton>
      </Tooltip>
    <td>
  </tr>
) : null}
```

- Note the `Mono` wrapper on the command text -- code (and machine-generated values) should be rendered with `Mono`
- You won't need the `display="block"` when the tooltip is in its own `td`.